File: JANUARY/11-01-2026/85.cs
public class Solution {
    public int MaximalRectangle(char[][] matrix) {
        if (matrix.Length == 0) return 0;

        int rows = matrix.Length, cols = matrix[0].Length;
        int[] heights = new int[cols];
        int ans = 0;

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                heights[j] = matrix[i][j] == '1' ? heights[j] + 1 : 0;
            }

            Stack<int> stack = new Stack<int>();
            for (int k = 0; k <= cols; k++) {
                int h = k == cols ? 0 : heights[k];
                while (stack.Count > 0 && h < heights[stack.Peek()]) {
                    int height = heights[stack.Pop()];
                    int width = stack.Count == 0 ? k : k - stack.Peek() - 1;
                    ans = Math.Max(ans, height * width);
                }
                stack.Push(k);
            }
        }

        return ans;
    }
}

File: JANUARY/11-01-2026/85.java
class Solution {
    private int largestRectangleArea(int[] heights) {
        Stack<Integer> stack = new Stack<>();
        int n = heights.length, ans = 0;
        for (int i = 0; i <= n; i++) {
            int h = (i == n) ? 0 : heights[i];
            while (!stack.isEmpty() && h < heights[stack.peek()]) {
                int height = heights[stack.pop()];
                int width = stack.isEmpty() ? i : i - stack.peek() - 1;
                ans = Math.max(ans, height * width);
            }
            stack.push(i);
        }
        return ans;
    }

    public int maximalRectangle(char[][] matrix) {
        if (matrix == null || matrix.length == 0) return 0;
        int rows = matrix.length, cols = matrix[0].length;
        int[] heights = new int[cols];
        int ans = 0;
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                heights[j] = matrix[i][j] == '1' ? heights[j] + 1 : 0;
            }
            ans = Math.max(ans, largestRectangleArea(heights));
        }
        return ans;
    }
}